qiskit/test/mock/fake_openpulse_2q.py

```diff
@@ -81,58 +81,124 @@ def __init__(self):
-                             sequence=[PulseQobjInstruction(name='fc', ch='d0',
-                                                            t0=0, phase='-P0*np.pi'),
-                                       PulseQobjInstruction(name='test_pulse_4', ch='d0', t0=0),
-                                       PulseQobjInstruction(name='fc', ch='d0',
-                                                            t0=0, phase='-P1*np.pi')]),
-                     Command(name='u2', qubits=[1],
-                             sequence=[PulseQobjInstruction(name='fc', ch='d1',
-                                                            t0=0, phase='-P0*np.pi'),
-                                       PulseQobjInstruction(name='test_pulse_4', ch='d1', t0=0),
-                                       PulseQobjInstruction(name='fc', ch='d1',
-                                                            t0=0, phase='-P0*np.pi')]),
-                     Command(name='u3', qubits=[0],
-                             sequence=[PulseQobjInstruction(name='test_pulse_1', ch='d0', t0=0)]),
-                     Command(name='u3', qubits=[1],
-                             sequence=[PulseQobjInstruction(name='test_pulse_3', ch='d1', t0=0)]),
-                     Command(name='cx', qubits=[0, 1],
-                             sequence=[PulseQobjInstruction(name='test_pulse_1', ch='d0', t0=0),
-                                       PulseQobjInstruction(name='test_pulse_2', ch='u0', t0=10),
-                                       PulseQobjInstruction(name='test_pulse_1', ch='d1', t0=20),
-                                       PulseQobjInstruction(name='fc', ch='d1',
-                                                            t0=20, phase=2.1)]),
-                     Command(name='ParametrizedGate', qubits=[0, 1],
-                             sequence=[PulseQobjInstruction(name='test_pulse_1', ch='d0', t0=0),
-                                       PulseQobjInstruction(name='test_pulse_2', ch='u0', t0=10),
-                                       PulseQobjInstruction(name='pv', ch='d1',
-                                                            t0=2, val='cos(P2)'),
-                                       PulseQobjInstruction(name='test_pulse_1', ch='d1', t0=20),
-                                       PulseQobjInstruction(name='fc', ch='d1',
-                                                            t0=20, phase=2.1)]),
-                     Command(name='measure', qubits=[0, 1],
-                             sequence=[PulseQobjInstruction(name='test_pulse_1', ch='m0', t0=0),
-                                       PulseQobjInstruction(name='test_pulse_1', ch='m1', t0=0),
-                                       PulseQobjInstruction(name='acquire', duration=10, t0=0,
-                                                            qubits=[0, 1], memory_slot=[0, 1])])]
-        )
+        self._defaults = PulseDefaults.from_dict({
```

Comment:
This is needed because `PulseDefaults` is still constructed using marshmallow. Marshmallow doesn't know how to interpret a `PulseLibraryitem` or a `PulseQobjInstruction` and serialize it for its internal representation now that they're not constructed using the schema. To avoid making this patch twice the size there were 2 ways I could think of workaround this, either teach marshmallow how to convert the new types to something it could use (something like #3181) or just make the constructor use `from_dict()`. The `from_dict` approach seemed easier to me, especially  because we're probably going to remove it from the provider classes soon too.

Reply:
Ah, so this is just temporary

Reply:
Yep, this will not be necessary once we convert everything away from using marshmallow. This will still work post marshmallow removal for ``PulseDefaults``, ``BackendConfiguration``, etc. But it's only needed here because we havn't converted everything yet. We can revert this piece once we do (it definitely makes it easier to work with as class constructors).